Fixes #19800: ModuleType import supports associating ModuleTypeProfile

jnovinger opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #19800
- Fixes issue by adding `profile` to the list of fields the import form should recognize
- Fixes up module type bulk import testing by adding necessary permissions for related models and adding inheritance of `BulkImportObjectsViewTestCase`
- Adds an additional regression assertion to make sure that this fix does not regress

<!--
    Please include a summary of the proposed changes below.
-->

jnovinger committed:
Fixes #19800: ModuleType import supports associating ModuleTypeProfile

jnovinger committed:
Fixes up ModuleTypeTestCase to include bulk import testing

Also includes an additional regression assertion.

jnovinger commented on netbox/dcim/tests/test_views.py:
```diff
@@ -1044,6 +1036,62 @@ def setUpTestData(cls):
+            InterfaceTemplate, FrontPortTemplate, RearPortTemplate, ModuleBayTemplate,
+        ]
+        for model in additional_permissions:
+            obj_perm.object_types.add(ObjectType.objects.get_for_model(model))
```
Open to suggestions on this approach.

jeremystretch commented on netbox/dcim/forms/bulk_import.py:
```diff
@@ -471,7 +471,7 @@ class Meta:
         fields = [
             'manufacturer', 'model', 'part_number', 'description', 'airflow', 'weight', 'weight_unit', 'comments',
-            'tags',
+            'tags', 'profile',
```
Could you move `profile` before `comments` in the list? It affects the ordering of the fields in the import form.

jnovinger commented on netbox/dcim/tests/test_views.py:
```diff
@@ -1044,6 +1036,62 @@ def setUpTestData(cls):
+        fan_module_type = ModuleType.objects.get(part_number='generic-fan')
+        fan_module_type_profile = ModuleTypeProfile.objects.get(name='Fan')
+
+        assert fan_module_type.profile == fan_module_type_profile
```
It turns out that the parent test _is_ actually only testing that the count is correct.

It feels like a FR to expand the scope of the parent test, especially when it could very well break a bunch of other tests.

Open to feedback on that, though.

jnovinger committed:
Address PR feedback

I ultimately left the extra asserts in for test_bulk_import_objects_with_permissionsince
since the parent test is currently only testing against number of
objects successfully imported. Will file a follow up FR to improve that
test.
